pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -874,13 +875,13 @@ func dnsZoneNotReadyMaybeTimedOut(cd *hivev1.ClusterDeployment, logger log.Field
 			controllerutils.UpdateConditionIfReasonOrMessageChange)
```

Comment:
If you go with the above suggestion (the `bool` return indicates the `cd` was updated) then we need to add a change check to this one as well, and return the `||` of the two.